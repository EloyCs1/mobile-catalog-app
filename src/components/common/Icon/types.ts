export enum IconName {
  MBST_LOGO,
  BAG_ACTIVE,
  BAG_INACTIVE,
}

export interface IconProps {
  testId?: string;
  name: IconName;
  width?: number;
  height?: number;
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}
