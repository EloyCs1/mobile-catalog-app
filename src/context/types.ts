import { ReactNode } from 'react';

export interface CatalogContextValue {
  favoriteFilter: boolean;
  toggleCatalogFilter: (favoriteFilter: boolean) => void;
  favoriteIds: number[];
  toggleCatalogId: (id: number) => void;
}

export interface CatalogProviderProps {
  children: ReactNode;
  values: CatalogContextValue;
}
