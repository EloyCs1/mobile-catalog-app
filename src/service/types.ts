export interface ErrorEntity {
  error: string;
  message: string;
}

export interface ProductListEntity {
  id: string;
  brand: string;
  name: string;
  basePrice: number;
  imageUrl: string;
}

export interface ProductEntity {
  id: string;
  brand: string;
  name: string;
  description: string;
  basePrice: number;
  rating: number;
  specs: {
    screen: string;
    resolution: string;
    processor: string;
    mainCamera: string;
    selfieCamera: string;
    battery: string;
    os: string;
    screenRefreshRate: string;
  };
  colorOptions: {
    name: string;
    hexCode: string;
    imageUrl: string;
  }[];
  storageOptions: {
    capacity: string;
    price: number;
  }[];
  similarProducts: {
    id: string;
    brand: string;
    name: string;
    basePrice: number;
    imageUrl: string;
  }[];
}
