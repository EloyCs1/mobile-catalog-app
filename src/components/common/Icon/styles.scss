.iconSpinner {
  transform-origin: center;
  animation: iconSpinner_svv 0.75s infinite linear;
}

.buttonIcon {
  background: none;
  border: none;
  padding: 0;
  cursor: pointer;
}

@keyframes iconSpinner_svv {
  100% {
    transform: rotate(360deg);
  }
}
