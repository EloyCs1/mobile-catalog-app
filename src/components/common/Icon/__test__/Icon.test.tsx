import { render } from "@testing-library/react";

import { AppWrapper } from "src/__test__/test.utils";
import Icon from "../Icon";
import { IconName } from "../types";

describe("Icon", () => {
  test("FAVORITES_FILLED render", () => {
    const wrapper = render(
      <AppWrapper>
        <Icon name={IconName.FAVORITES_FILLED} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("FAVORITES_OUTLINED render", () => {
    const wrapper = render(
      <AppWrapper>
        <Icon name={IconName.FAVORITES_OUTLINED} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("MARVEL_LOGO render", () => {
    const onClick = jest.fn();
    const wrapper = render(
      <AppWrapper>
        <Icon name={IconName.MARVEL_LOGO} onClick={onClick} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("SEARCH render", () => {
    const wrapper = render(
      <AppWrapper>
        <Icon name={IconName.SEARCH} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("SPINNER render", () => {
    const wrapper = render(
      <AppWrapper>
        <Icon name={IconName.SPINNER} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
