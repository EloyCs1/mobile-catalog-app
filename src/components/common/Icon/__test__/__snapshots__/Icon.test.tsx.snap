// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon FAVORITES_FILLED render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        data-testid="icon"
        fill="none"
        height="21.68"
        viewBox="0 0 24 22"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12 3.80348L6 0.161865L0 3.80348V11.607L12 21.8382L24 11.607V3.80348L18 0.161865L12 3.80348Z"
          fill="#EC1D24"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      data-testid="icon"
      fill="none"
      height="21.68"
      viewBox="0 0 24 22"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 3.80348L6 0.161865L0 3.80348V11.607L12 21.8382L24 11.607V3.80348L18 0.161865L12 3.80348Z"
        fill="#EC1D24"
        fill-rule="evenodd"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Icon FAVORITES_OUTLINED render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        data-testid="icon"
        fill="none"
        height="21.68"
        viewBox="0 0 15 14"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.75 1.55237L5.26885 0.697502L4.75 0.382594L4.23115 0.697502L4.75 1.55237ZM7.75 3.37318L7.23115 4.22804L7.75 4.54295L8.26885 4.22804L7.75 3.37318ZM1.75 3.37318L1.23115 2.51831L0.75 2.81034V3.37318H1.75ZM1.75 7.27491H0.75V7.73644L1.1012 8.03587L1.75 7.27491ZM7.75 12.3905L7.1012 13.1515L7.75 13.7046L8.3988 13.1515L7.75 12.3905ZM13.75 7.27491L14.3988 8.03587L14.75 7.73644V7.27491H13.75ZM13.75 3.37318H14.75V2.81034L14.2688 2.51831L13.75 3.37318ZM10.75 1.55237L11.2689 0.697502L10.75 0.382594L10.2312 0.697502L10.75 1.55237ZM4.23115 2.40723L7.23115 4.22804L8.26885 2.51831L5.26885 0.697502L4.23115 2.40723ZM2.26885 4.22804L5.26885 2.40723L4.23115 0.697502L1.23115 2.51831L2.26885 4.22804ZM2.75 7.27491V3.37318H0.75V7.27491H2.75ZM8.3988 11.6296L2.3988 6.51395L1.1012 8.03587L7.1012 13.1515L8.3988 11.6296ZM8.3988 13.1515L14.3988 8.03587L13.1012 6.51395L7.1012 11.6296L8.3988 13.1515ZM14.75 7.27491V3.37318H12.75V7.27491H14.75ZM14.2688 2.51831L11.2689 0.697502L10.2312 2.40723L13.2312 4.22804L14.2688 2.51831ZM10.2312 0.697502L7.23115 2.51831L8.26885 4.22804L11.2689 2.40723L10.2312 0.697502Z"
          fill="white"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      data-testid="icon"
      fill="none"
      height="21.68"
      viewBox="0 0 15 14"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.75 1.55237L5.26885 0.697502L4.75 0.382594L4.23115 0.697502L4.75 1.55237ZM7.75 3.37318L7.23115 4.22804L7.75 4.54295L8.26885 4.22804L7.75 3.37318ZM1.75 3.37318L1.23115 2.51831L0.75 2.81034V3.37318H1.75ZM1.75 7.27491H0.75V7.73644L1.1012 8.03587L1.75 7.27491ZM7.75 12.3905L7.1012 13.1515L7.75 13.7046L8.3988 13.1515L7.75 12.3905ZM13.75 7.27491L14.3988 8.03587L14.75 7.73644V7.27491H13.75ZM13.75 3.37318H14.75V2.81034L14.2688 2.51831L13.75 3.37318ZM10.75 1.55237L11.2689 0.697502L10.75 0.382594L10.2312 0.697502L10.75 1.55237ZM4.23115 2.40723L7.23115 4.22804L8.26885 2.51831L5.26885 0.697502L4.23115 2.40723ZM2.26885 4.22804L5.26885 2.40723L4.23115 0.697502L1.23115 2.51831L2.26885 4.22804ZM2.75 7.27491V3.37318H0.75V7.27491H2.75ZM8.3988 11.6296L2.3988 6.51395L1.1012 8.03587L7.1012 13.1515L8.3988 11.6296ZM8.3988 13.1515L14.3988 8.03587L13.1012 6.51395L7.1012 11.6296L8.3988 13.1515ZM14.75 7.27491V3.37318H12.75V7.27491H14.75ZM14.2688 2.51831L11.2689 0.697502L10.2312 2.40723L13.2312 4.22804L14.2688 2.51831ZM10.2312 0.697502L7.23115 2.51831L8.26885 4.22804L11.2689 2.40723L10.2312 0.697502Z"
        fill="white"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Icon MARVEL_LOGO render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="buttonIcon"
        data-testid="icon-button"
      >
        <svg
          data-testid="icon"
          fill="none"
          height="21.68"
          viewBox="0 0 130 52"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_14854_731)"
          >
            <path
              d="M130 0H0V52H130V0Z"
              fill="#EC1D24"
            />
            <path
              d="M126.222 40.059V47.965H111.58V4H119.465V40.059H126.222ZM63.658 25.559C63.048 25.853 62.41 25.999 61.788 26.001V11.861H61.828C62.45 11.856 67.092 12.045 67.092 18.854C67.092 22.413 65.512 24.658 63.658 25.559ZM40.55 34.24L42.733 15.441L44.998 34.24H40.55ZM110.205 12.025V4.007H87.879L84.204 30.786L80.574 4.006H72.522L73.423 11.156C72.495 9.324 69.199 4.006 61.943 4.006C61.896 4.004 53.883 4.006 53.883 4.006L53.852 43.038L47.984 4.007L37.439 4.002L31.367 44.442L31.369 4.007H21.278L17.64 26.724L14.096 4.006H4V47.972H11.95V26.78L15.568 47.972H19.794L23.359 26.78V47.972H38.686L39.614 41.21H45.784L46.711 47.972L61.758 47.98H61.768V47.972H61.788V33.702L63.633 33.432L67.45 47.982H75.234L75.232 47.972H75.254L70.243 30.924C72.781 29.044 75.649 24.28 74.886 19.721V19.719C74.894 19.777 79.615 48 79.615 48L88.871 47.973L95.198 8.123V47.973H110.205V40.065H103.081V29.985H110.205V21.955H103.081V12.024L110.205 12.025Z"
              fill="#FEFEFE"
            />
            <path
              d="M0 0H30V52H0V0Z"
              fill="#EC1D24"
            />
            <path
              d="M31.5 48V4H21.291L17.651 26.735L14.102 4H4V48H12V26.792L15.577 48H19.806L23.374 26.792V48H31.5Z"
              fill="#FEFEFE"
            />
          </g>
          <defs>
            <clippath
              id="clip0_14854_731"
            >
              <rect
                fill="white"
                height="52"
                width="130"
              />
            </clippath>
          </defs>
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="buttonIcon"
      data-testid="icon-button"
    >
      <svg
        data-testid="icon"
        fill="none"
        height="21.68"
        viewBox="0 0 130 52"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_14854_731)"
        >
          <path
            d="M130 0H0V52H130V0Z"
            fill="#EC1D24"
          />
          <path
            d="M126.222 40.059V47.965H111.58V4H119.465V40.059H126.222ZM63.658 25.559C63.048 25.853 62.41 25.999 61.788 26.001V11.861H61.828C62.45 11.856 67.092 12.045 67.092 18.854C67.092 22.413 65.512 24.658 63.658 25.559ZM40.55 34.24L42.733 15.441L44.998 34.24H40.55ZM110.205 12.025V4.007H87.879L84.204 30.786L80.574 4.006H72.522L73.423 11.156C72.495 9.324 69.199 4.006 61.943 4.006C61.896 4.004 53.883 4.006 53.883 4.006L53.852 43.038L47.984 4.007L37.439 4.002L31.367 44.442L31.369 4.007H21.278L17.64 26.724L14.096 4.006H4V47.972H11.95V26.78L15.568 47.972H19.794L23.359 26.78V47.972H38.686L39.614 41.21H45.784L46.711 47.972L61.758 47.98H61.768V47.972H61.788V33.702L63.633 33.432L67.45 47.982H75.234L75.232 47.972H75.254L70.243 30.924C72.781 29.044 75.649 24.28 74.886 19.721V19.719C74.894 19.777 79.615 48 79.615 48L88.871 47.973L95.198 8.123V47.973H110.205V40.065H103.081V29.985H110.205V21.955H103.081V12.024L110.205 12.025Z"
            fill="#FEFEFE"
          />
          <path
            d="M0 0H30V52H0V0Z"
            fill="#EC1D24"
          />
          <path
            d="M31.5 48V4H21.291L17.651 26.735L14.102 4H4V48H12V26.792L15.577 48H19.806L23.374 26.792V48H31.5Z"
            fill="#FEFEFE"
          />
        </g>
        <defs>
          <clippath
            id="clip0_14854_731"
          >
            <rect
              fill="white"
              height="52"
              width="130"
            />
          </clippath>
        </defs>
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Icon SEARCH render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        data-testid="icon"
        fill="none"
        height="21.68"
        viewBox="0 0 13 13"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
          fill="black"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      data-testid="icon"
      fill="none"
      height="21.68"
      viewBox="0 0 13 13"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
        fill="black"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Icon SPINNER render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        data-testid="icon"
        height="21.68"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="iconSpinner"
          d="M10.14,1.16a11,11,0,0,0-9,8.92A1.59,1.59,0,0,0,2.46,12,1.52,1.52,0,0,0,4.11,10.7a8,8,0,0,1,6.66-6.61A1.42,1.42,0,0,0,12,2.69h0A1.57,1.57,0,0,0,10.14,1.16Z"
          fill="black"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      data-testid="icon"
      height="21.68"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="iconSpinner"
        d="M10.14,1.16a11,11,0,0,0-9,8.92A1.59,1.59,0,0,0,2.46,12,1.52,1.52,0,0,0,4.11,10.7a8,8,0,0,1,6.66-6.61A1.42,1.42,0,0,0,12,2.69h0A1.57,1.57,0,0,0,10.14,1.16Z"
        fill="black"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
